src/snowflake/cli/plugins/nativeapp/codegen/snowpark/python_processor.py

```diff
@@ -149,22 +152,27 @@ def __init__(
         project_definition: NativeApp,
         project_root: Path,
         deploy_root: Path,
+        generated_root: Path,
     ):
         super().__init__(
             project_definition=project_definition,
             project_root=project_root,
             deploy_root=deploy_root,
+            generated_root=generated_root,
         )
         self.project_definition = project_definition
         self.project_root = project_root
         self.deploy_root = deploy_root
+        self.generated_root = generated_root
 
     def process(
         self,
         artifact_to_process: PathMapping,
         processor_mapping: Optional[ProcessorMapping],
         **kwargs,
-    ) -> str:  # String output is temporary until we have better e2e testing mechanism
+    ) -> Dict[
```

Comment:
I was just waiting for the file output to remove the hack. I don't think we should do this. First, it breaks all existing snapshots, so it's hard to check for regressions. Second, it doesn't let us test the actual output anyway, so it's of very limited value.